### PR TITLE
IntelFsp2WrapperPkg: Save FspHobListPtr Right After FspMemoryInit Exits

### PR DESCRIPTION
**Pull request by dulin0**
# Description

Save FspHobList pointer to HOB right after FspMemoryInit exits so that FspHobList pointer is available when performing platform related reset in CallFspWrapperResetSystem(). Some platforms may consume FSP HOBs prior to performing platform related reset.

- [ ] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.

## How This Was Tested

This PR was tested with FSP-M, FSP-S, FSP multi-phase memory initialization and FSP multi-phase silicon initialization enabled. And it's confirmed in the test that the bootloader can successfully get and save the FspHobList pointer in different phases.

## Integration Instructions
N/A

### CODE REVIEW

File: IntelFsp2WrapperPkg/Library/FspWrapperMultiPhaseProcessLib/PeiFspWrapperMultiPhaseProcessLib.c [hunk 1]
```diff
@@ -352,6 +356,15 @@ FspWrapperMultiPhaseHandler (
     FspMultiPhaseParams.MultiPhaseParamPtr = NULL;
     Status                                 = CallFspMultiPhaseEntry (&FspMultiPhaseParams, FspHobListPtr, ComponentIndex);
 
+    //
+    // Refresh FspHobList pointer stored in HOB.
```
Comment (ChaselChiu) [inline IntelFsp2WrapperPkg/Library/FspWrapperMultiPhaseProcessLib/PeiFspWrapperMultiPhaseProcessLib.c:360]: When the platforms only support VariableRequest, we will need similar code to refresh FspHobList in FspWrapperVariableRequestHandler() too. (to refresh the pointer after FspHobList point migrated to permanent memory)

Comment (dulin0) [reply IntelFsp2WrapperPkg/Library/FspWrapperMultiPhaseProcessLib/PeiFspWrapperMultiPhaseProcessLib.c:371]: Done. Thanks for pointing this out.

File: IntelFsp2WrapperPkg/Library/FspWrapperMultiPhaseProcessLib/PeiFspWrapperMultiPhaseProcessLib.c [hunk 2]
```diff
@@ -288,6 +292,15 @@ FspWrapperVariableRequestHandler (
     }
   }
 
+  //
+  // Refresh FspHobList pointer stored in HOB.
+  //
+  GuidHob = GetFirstGuidHob (&gFspHobGuid);
+  if (GuidHob != NULL) {
```
Comment (niruiyu) [inline IntelFsp2WrapperPkg/Library/FspWrapperMultiPhaseProcessLib/PeiFspWrapperMultiPhaseProcessLib.c:299]: can we add an assertion ASSERT (GuidHob != NULL)?
If-check is ok to be kept.

Comment (dulin0) [reply IntelFsp2WrapperPkg/Library/FspWrapperMultiPhaseProcessLib/PeiFspWrapperMultiPhaseProcessLib.c]: Done. Thanks for the comment.

**Review by niruiyu — APPROVED** [review]

**Review by AshrafAliS — APPROVED** [review]

**Pull request merged**